pkg/reconcile/common.go

```diff
@@ -140,16 +141,37 @@ func (u *MonitorResourceCommon) UpdateMonitorResourceStatus(cr client.Object) (r
 	return reconcile.StopReconcile()
 }
 
-func (u *MonitorResourceCommon) GetClusterID() string {
-	if u.ClusterID == "" {
-		var version configv1.ClusterVersion
-		err := u.Client.Get(u.Ctx, client.ObjectKey{Name: "version"}, &version)
-		if err != nil {
-			return ""
-		}
-		u.ClusterID = string(version.Spec.ClusterID)
+// GetOSDClusterID returns the ID for the cluster based on its ClusterVersion
+func (u *MonitorResourceCommon) GetOSDClusterID() (string, error) {
+	var version configv1.ClusterVersion
+	err := u.Client.Get(u.Ctx, client.ObjectKey{Name: "version"}, &version)
+	if err != nil {
+		return "", err
 	}
-	return u.ClusterID
```

Comment:
Caching the cluster ID probably made sense when RMO only acted on a single cluster, but now that it's being adapted for HyperShift and potentially acting on different clusters with each reconcile, we should avoid caching metadata between executions.